source/grid/grid_in.cc

```diff
@@ -1615,11 +1615,14 @@ GridIn<dim, spacedim>::read_comsol_mphtxt(std::istream &in)
   whole_file >> n_types;
   for (unsigned int type = 0; type < n_types; ++type)
     {
-      std::string object_type;
+      // The object type is prefixed by the number of characters the
+      // object type string takes up (e.g., 3 for 'tri' and 5 for
+      // 'prism'), but we really don't need that.
       {
         unsigned int dummy;
         whole_file >> dummy;
       }
+      std::string object_type;
```

Comment:
Would you be willing to write a conversion function from this to `ReferenceCell`? That would get rid of a lot of string comparisons and make the code more generic.

Reply:
Actually, I'm not sure that's true. The names the COMSOL format uses are not generic. For example, they use "prism" instead of our "wedge". But it would be easy enough to have a table at the top of the function that does that conversion via a `std::map`. Would that work?

Reply:
@drwells  Maybe like in the patch I just pushed?

Reply:
Yes, this is the kind of change I had in mind.